### PR TITLE
window app changes for MG12 and MG24

### PR DESCRIPTION
**Pull request by ShubhamMalasane**
Window app bring up
* Created a separate app task
* Fix the chip die error
* Fix critical nesting error

Change overview
* Created separate app task and call init function call inside that task earlier it is called in main function
* To resolve critical nesting issue uses vTaskSuspend function inside newly created task.

Testing
* Perform core cluster test cases 
* Perform sanity test cases
* Perform all test cases which is required cluster for window app

### CODE REVIEW

**Review by jepenven-silabs — CHANGES_REQUESTED** [review]
Don't have time to review right now but will later

File: examples/window-app/efr32/src/WindowAppImpl.cpp [hunk 1]
```diff
@@ -336,12 +370,16 @@ void WindowAppImpl::DispatchEvent(const WindowApp::Event & event)
         break;
     case EventId::CoverChange:
         mIconTimer.Start();
+        #ifdef DISPLAY_ENABLED
```
Comment (jmeg-sfy) [inline examples/window-app/efr32/src/WindowAppImpl.cpp:373]: Actually there is another PR doing this display enable handling
please wait for it to be merged first and rebase

**Comment by jepenven-silabs** [comment]
This PR goes against all of the logic in the Window App by destroying the abstraction layer it implemented.

Either streamline the complete example to match other example with an AppTask and a BaseApplication class or leave this examples as it is.

@rcasallas-silabs any thoughts on this ?

File: examples/window-app/common/src/WindowApp.cpp [hunk 2]
```diff
@@ -109,6 +109,8 @@ WindowApp::Cover * WindowApp::GetCover(chip::EndpointId endpoint)
     return nullptr;
 }
 
+CHIP_ERROR WindowApp::StartAppTask() { return CHIP_NO_ERROR;}
```
Comment (jepenven-silabs) [inline examples/window-app/common/src/WindowApp.cpp:112]: This doesn't do anything and should be removed

File: examples/window-app/efr32/src/WindowAppImpl.cpp [hunk 3]
```diff
@@ -17,6 +17,7 @@
 
 #include <AppConfig.h>
 #include <LcdPainter.h>
+//#include <WindowApp.h>
```
Comment (jepenven-silabs) [inline examples/window-app/efr32/src/WindowAppImpl.cpp:23]: Why would you remove the include to the base class ?

File: examples/window-app/efr32/src/WindowAppImpl.cpp [hunk 4]
```diff
@@ -122,7 +128,9 @@ void WindowAppImpl::Timer::TimerCallback(TimerHandle_t xTimer)
 //------------------------------------------------------------------------------
 
 StackType_t sAppStack[APP_TASK_STACK_SIZE / sizeof(StackType_t)];
+StackType_t sAppStack1[APP_TASK_STACK_SIZE + 2048/ sizeof(StackType_t)];
```
Comment (jepenven-silabs) [inline examples/window-app/efr32/src/WindowAppImpl.cpp:133]: Magic number and why doubling everything?

File: examples/window-app/efr32/src/WindowAppImpl.cpp [hunk 5]
```diff
@@ -122,7 +128,9 @@ void WindowAppImpl::Timer::TimerCallback(TimerHandle_t xTimer)
...
 StaticTask_t sAppTaskStruct;
+StaticTask_t sAppTaskStruct1;
```
Comment (jepenven-silabs) [inline examples/window-app/efr32/src/WindowAppImpl.cpp:135]: But why duplicating and appending a 1?

File: examples/window-app/efr32/src/WindowAppImpl.cpp [hunk 6]
```diff
@@ -151,6 +159,24 @@ void WindowAppImpl::OnIconTimeout(WindowApp::Timer & timer)
...
+CHIP_ERROR WindowAppImpl::StartAppTask()
+{
+
+    // Start App task.
+    TaskHandle_t a = xTaskCreateStatic(AppTaskMain, APP_TASK_NAME, ArraySize(sAppStack1), NULL, 1, sAppStack1, &sAppTaskStruct1);
+    return (a == nullptr) ? CHIP_ERROR_NO_MEMORY : CHIP_NO_ERROR;
+}
+
+void WindowAppImpl::AppTaskMain(void * pvParameter){
+	CHIP_ERROR err = sInstance.Init();
```
Comment (jepenven-silabs) [inline examples/window-app/efr32/src/WindowAppImpl.cpp:175]: This main function doesn't do anything .

File: examples/window-app/efr32/src/WindowAppImpl.cpp [hunk 7]
```diff
@@ -164,8 +190,21 @@ CHIP_ERROR WindowAppImpl::Init()
...
+
+    // Initialize LEDs
+    LEDWidget::InitGpio();
+    mStatusLED.Init(APP_STATE_LED);
+    mActionLED.Init(APP_ACTION_LED);
+
+#ifdef DISPLAY_ENABLED
+    slLCD.Init();
 #endif
-    WindowApp::Init();
```
Comment (jepenven-silabs) [inline examples/window-app/efr32/src/WindowAppImpl.cpp:168]: That will break the OpenThread implementation

File: examples/window-app/efr32/src/WindowAppImpl.cpp [hunk 8]
```diff
@@ -183,15 +222,6 @@ CHIP_ERROR WindowAppImpl::Init()
         return CHIP_ERROR_NO_MEMORY;
     }
 
-    // Initialize LEDs
-    LEDWidget::InitGpio();
-    mStatusLED.Init(APP_STATE_LED);
-    mActionLED.Init(APP_ACTION_LED);
-
-#ifdef DISPLAY_ENABLED
-    slLCD.Init();
```
Comment (jepenven-silabs) [inline examples/window-app/efr32/src/WindowAppImpl.cpp:192]: That will break OpenThread implementation

**Review by jepenven-silabs — CHANGES_REQUESTED** [review]
This whole PR needs to be redone.

**Comment by jepenven-silabs** [comment]
Closing considering the amount of rework needed.

**Pull request closed without merging**